### PR TITLE
Added side table for cutsets

### PR DESCRIPTION
**Pull request by Kasmadei**
Resolves: https://github.com/kbss-cvut/fta-fmea-ui/issues/1

### CODE REVIEW

File: src/components/editor/faultTree/menu/FaultEventScenariosTable.tsx [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+
+  return (
+    <Box className={classes.tableContainer}>
+      {/* Table Head */}
+      <TableContainer>
+        <Table>
+          <TableHead>
+            <TableRow>
+              <TableCell>Cutset</TableCell>
+              <TableCell>Probability</TableCell>
```
Comment (blcham) [inline src/components/editor/faultTree/menu/FaultEventScenariosTable.tsx:71]: missing i18n

**Review by blcham — CHANGES_REQUESTED** [review]
See my comments.

File: src/components/editor/faultTree/menu/FaultEventScenariosTable.tsx [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+      const parts = scenarioPart.iri.split("/");
+      const lastPart = parts[parts.length - 1];
+      cutsetsStrings.push(lastPart);
+    }
+  } else {
+    // scenarioParts is an object
+    const parts = item.scenarioParts.iri.split("/");
+    const lastPart = parts[parts.length - 1];
+    cutsetsStrings.push(lastPart);
+  }
```
[the remark this answers is not in the document]
Comment (Kasmadei) [reply src/components/editor/faultTree/menu/FaultEventScenariosTable.tsx:33]: Changed (created sep. function for this)

File: src/components/editor/faultTree/Editor.tsx [hunk 3]
```diff
@@ -51,7 +52,8 @@ const Editor = ({ setAppBarName }: DashboardTitleProps) => {
       setAppBarName(faultTree.name);
       setRootEvent(faultTree.manifestingEvent);
       if (faultTree.faultEventScenarios) {
-        setFaultEventScenarios(getScenariosWithHighestProbability(faultTree.faultEventScenarios));
+        const result = getScenariosWithHighestProbability(faultTree.faultEventScenarios);
```
[the remark this answers is not in the document]
Comment (Kasmadei) [reply src/components/editor/faultTree/Editor.tsx:55]: Right, i have changed it already in next PR

File: src/components/editor/faultTree/menu/FaultEventScenariosTable.tsx (hunk 1 above)
Comment (Kasmadei) [reply src/components/editor/faultTree/menu/FaultEventScenariosTable.tsx]: Fixed

File: src/components/editor/faultTree/menu/FaultEventScenariosTable.tsx [hunk 4]
```diff
@@ -0,0 +1,105 @@
+import * as React from "react";
+import { useEffect, useState, FC } from "react";
+import { Box, Typography, useTheme } from "@mui/material";
+import { FaultEventScenario } from "../../../../models/faultEventScenario";
+import { Table, TableBody, TableCell, TableContainer, TableHead, TableRow, Paper } from "@mui/material";
+import useStyles from "./FaultEventScenariosTable.styles";
+import { asArray } from "@utils/treeUtils";
```
Comment (blcham) [inline src/components/editor/faultTree/menu/FaultEventScenariosTable.tsx:7]: Although it is confusing that it belongs to treeUtils as it does not have anything to do with trees, but it is acceptable to me like that. I would rather add it to a new class called CollectionsUtils, ArrayUtils or just Utils.

**Pull request merged**